### PR TITLE
User Authentication

### PR DESCRIPTION
**Pull request by syuanivy**
1. Create User model with OneToOne association with Person. Migration needed!  
2. Add user(required), bio, location, birth_date attributes to Person
3. Add user creation, login and logout endpoint

### CODE REVIEW

File: webrest/models/person.py [hunk 1]
```diff
@@ -40,12 +48,28 @@ class Person(models.Model):
...
                                null=False,
                                max_length=512)
 
+    bio = models.TextField(max_length=500, null=True, blank=True)
+
+    location = models.CharField(max_length=30, null=True, blank=True)
+
+    birth_date = models.DateField(null=True, blank=True)
+
+    @receiver(post_save, sender=User)
```
Comment (neptscorpio) [inline webrest/models/person.py]: Curious: what are these receivers used for?

File: webrest/views.py [hunk 2]
```diff
@@ -10,20 +10,66 @@
...
+
+        return Response({'status': 'success', 'person_id': new_person.id}, 201)
+
+    @list_route(methods=['post'])
+    def authenticate(self, request):
+        username = request.data['username']
+        password = request.data['password']
+        user = authenticate(username=username, password=password)
+
+        if user is not None:
```
Comment (neptscorpio) [inline webrest/views.py]: a more pythonish way:   "if user:"

File: webrest/views.py [hunk 3]
```diff
@@ -10,20 +10,66 @@
...
+            login(request, user)
```
Comment (neptscorpio) [inline webrest/views.py]: what is "login" used for?

File: webrest/views.py [hunk 4]
```diff
@@ -10,20 +10,66 @@
...
+            return Response(
+                {'status': 'success', 'message': 'User is successfully logged in.', 'person_id': user.person.id}, 202)
```
Comment (neptscorpio) [inline webrest/views.py]: from rest_framework import status
use  "status.HTTP_202_ACCEPTED" to replace hard code "202"

File: webrest/models/person.py (hunk 1 above)
Comment (neptscorpio) [inline webrest/models/person.py]: These are call back functions to unsure Person gets created whenever User is created.
That's why sender = User.
Same for update.

File: webrest/views.py (hunk 3 above)
Comment (syuanivy) [inline webrest/views.py]: session

**Pull request merged**